tests/unit_tests/core/util/byte_stream.cc

```diff
@@ -50,6 +50,93 @@ struct ByteStreamFixture
+  BOOST_CHECK_THROW(input.ConsumeData(1), std::length_error);
+  BOOST_CHECK_THROW(input.ReadBytes(1), std::length_error);
+  BOOST_CHECK_THROW(input.ReadUInt8(), std::length_error);
+  BOOST_CHECK_THROW(input.ReadUInt16(), std::length_error);
+  BOOST_CHECK_THROW(input.ReadUInt32(), std::length_error);
+  BOOST_CHECK_THROW(input.ReadUInt64(), std::length_error);
+}
+
+BOOST_AUTO_TEST_CASE(InputByteStream)
+{
+  core::InputByteStream input(m_IPv4Array.data(), m_IPv4Array.size());
+  BOOST_CHECK_NO_THROW(input.ConsumeData(0));
+  BOOST_CHECK_EQUAL(input.ReadUInt8(), m_IPv4Array.at(0));
+  BOOST_CHECK_EQUAL(input.ReadBytes(3), &m_IPv4Array.at(1));
+  BOOST_CHECK_THROW(input.ConsumeData(1), std::length_error);
+}
+
+BOOST_AUTO_TEST_CASE(OutputByteStream)
+{
+  std::array<std::uint8_t, 4> buffer;
+  core::OutputByteStream output(buffer.data(), buffer.size());
+  BOOST_CHECK_NO_THROW(output.WriteData(nullptr, 0));
+  BOOST_CHECK_NO_THROW(output.WriteUInt8(m_IPv4Array.at(0)));
+  BOOST_CHECK_EQUAL(output.GetSize(), 4);
+  BOOST_CHECK_EQUAL(output.GetData(), buffer.data());
+  BOOST_CHECK_EQUAL(output.GetPosition(), buffer.data() + 1);
+  BOOST_CHECK_NO_THROW(output.WriteData(&m_IPv4Array.at(1), 3));
+  BOOST_CHECK_EQUAL(output.GetPosition(), buffer.data() + 4);
+  BOOST_CHECK_THROW(output.WriteUInt8(1), std::length_error);
+  BOOST_CHECK_EQUAL_COLLECTIONS(
+      buffer.data(),
+      buffer.data() + buffer.size(),
+      m_IPv4Array.data(),
+      m_IPv4Array.data() + m_IPv4Array.size());
+}
+
+BOOST_AUTO_TEST_CASE(Bits16Test)
+{
+  std::uint16_t value = 256;
+  std::size_t size = sizeof(std::uint16_t);
+  std::uint8_t buffer[sizeof(std::uint16_t)];
+  core::OutputByteStream output(buffer, size);
+  BOOST_CHECK_NO_THROW(output.WriteUInt16(value));
+  BOOST_CHECK_THROW(output.WriteUInt8(0), std::length_error);
+  core::InputByteStream input(buffer, size);
+  BOOST_CHECK_EQUAL(input.ReadUInt16(), value);
+  BOOST_CHECK_THROW(input.ReadUInt8(), std::length_error);
+}
+
+BOOST_AUTO_TEST_CASE(Bits32Test)
+{
+  std::uint32_t value = 65536;
+  std::size_t size = sizeof(std::uint32_t);
+  std::uint8_t buffer[sizeof(std::uint32_t)];
+  core::OutputByteStream output(buffer, size);
+  BOOST_CHECK_NO_THROW(output.WriteUInt32(value));
+  BOOST_CHECK_THROW(output.WriteUInt8(0), std::length_error);
+  core::InputByteStream input(buffer, size);
+  BOOST_CHECK_EQUAL(input.ReadUInt32(), value);
+  BOOST_CHECK_THROW(input.ReadUInt8(), std::length_error);
+}
+
+BOOST_AUTO_TEST_CASE(Bits64Test)
+{
+  std::uint64_t value = 16777216;
+  std::size_t size = sizeof(std::uint64_t);
+  std::uint8_t buffer[sizeof(std::uint64_t)];
+  core::OutputByteStream output(buffer, size);
+  BOOST_CHECK_NO_THROW(output.WriteUInt64(value));
+  BOOST_CHECK_THROW(output.WriteUInt8(0), std::length_error);
+  core::InputByteStream input(buffer, size);
+  BOOST_CHECK_EQUAL(input.ReadUInt64(), value);
+  BOOST_CHECK_THROW(input.ReadUInt8(), std::length_error);
```

Comment:
For the sake of simplicity, how about something like:

```cpp
diff --git a/tests/unit_tests/core/util/byte_stream.cc b/tests/unit_tests/core/util/byte_stream.cc
index 7981de8e..5ab88c53 100644
--- a/tests/unit_tests/core/util/byte_stream.cc
+++ b/tests/unit_tests/core/util/byte_stream.cc
@@ -33,6 +33,9 @@
 
 #include <boost/test/unit_test.hpp>
 
+#include <array>
+#include <limits>
+
 #include "core/util/byte_stream.h"
 
 namespace core = kovri::core;
@@ -126,13 +129,14 @@ BOOST_AUTO_TEST_CASE(Bits32Test)
 
 BOOST_AUTO_TEST_CASE(Bits64Test)
 {
-  std::uint64_t value = 16777216;
-  std::size_t size = sizeof(std::uint64_t);
-  std::uint8_t buffer[sizeof(std::uint64_t)];
-  core::OutputByteStream output(buffer, size);
+  std::array<std::uint8_t, sizeof(std::uint64_t)> buffer {{}};
+  std::uint64_t value = std::numeric_limits<std::uint64_t>::max();
+
+  core::OutputByteStream output(buffer.data(), buffer.size());
   BOOST_CHECK_NO_THROW(output.WriteUInt64(value));
   BOOST_CHECK_THROW(output.WriteUInt8(0), std::length_error);
-  core::InputByteStream input(buffer, size);
+
+  core::InputByteStream input(buffer.data(), buffer.size());
   BOOST_CHECK_EQUAL(input.ReadUInt64(), value);
   BOOST_CHECK_THROW(input.ReadUInt8(), std::length_error);
 }
```

and refactor where appropriate for the previous cases?